src/main/java/net/sf/jabref/gui/NewFileDialogs.java

```diff
@@ -0,0 +1,161 @@
+            }
+            super.approveSelection();
+        }
+    };
+
+    private final JFrame parent;
+    private final String directory;
+    private FileNameExtensionFilter extFilter;
+    private Set<FileExtensions> extensions = EnumSet.noneOf(FileExtensions.class);
+
+
+    public NewFileDialogs(JFrame owner) {
+        this(owner, getWorkingDir());
+    }
+
+    public NewFileDialogs(JFrame owner, String dir) {
+        this.parent = owner;
+        this.directory = dir;
+        System.out.println("DIR in Constructor" + Paths.get(dir));
+        fileChooser.setCurrentDirectory(Paths.get(dir).toFile());
+    }
+
+    private void saveFileDialog() {
+        fileChooser.setDialogType(JFileChooser.SAVE_DIALOG);
+    }
+
+    private void openFileDialog() {
+        fileChooser.setDialogType(JFileChooser.OPEN_DIALOG);
+
+    }
+
+    public NewFileDialogs dirsOnly() {
+        fileChooser.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+        return this;
+    }
+
+    public NewFileDialogs withExtension(FileExtensions singleExt) {
+        withExtension(EnumSet.of(singleExt));
+        return this;
+    }
+
+    public NewFileDialogs withExtension(Set<FileExtensions> fileExtensions) {
+        this.extensions = fileExtensions;
+
+        for (FileExtensions ext : fileExtensions) {
+            extFilter = new FileNameExtensionFilter(ext.getDescription(), ext.getExtensions());
+            fileChooser.addChoosableFileFilter(extFilter);
+
+        }
+
+        return this;
+    }
+
+    public NewFileDialogs updateWorkingDirectory() {
+        updateWorkingDirectorySetting(this.directory);
+        return this;
+    }
+
+    public List<String> getMultipleFileNames() {
+        openFileDialog();
+        fileChooser.setMultiSelectionEnabled(true);
+
+        if (showDialogAndIsAccepted()) {
+            List<String> files = Arrays.stream(fileChooser.getSelectedFiles()).map(f -> f.toString())
+                    .collect(Collectors.toList());
+
+            return files;
+        }
+
+        return Collections.emptyList();
+
+    }
+
+    public Path getSelectedFile() {
+        openFileDialog();
+        if (showDialogAndIsAccepted()) {
+            return fileChooser.getSelectedFile().toPath();
+        }
+        return Paths.get("");
+    }
+
+    public Path saveNewFile() {
+        saveFileDialog();
+        if (showDialogAndIsAccepted()) {
+            File f = fileChooser.getSelectedFile();
+            if (!extensions.isEmpty() && !fileChooser.accept(f)) {
+
+                System.out.println("Exts " + extensions);
+
+                return new File(f.getPath() + extensions.iterator().next().getFirstExtensionWithDot()).toPath();
+            }
+            return f.toPath();
+        }
+        return Paths.get("");
+    }
+
+    private boolean showDialogAndIsAccepted() {
+        return fileChooser.showDialog(parent, null) == JFileChooser.APPROVE_OPTION;
+    }
+
+    private static String getWorkingDir() {
```

Comment:
I agree that it is simple here, but I'm still under the impression that it indeed is OK. With that said, I also happens to inject it in gui code every now and then.

I guess here one may actually inject the preference value rather than the complete preferences as well?